### PR TITLE
 New API definitions to improve NotificationTemplateManager interface

### PR DESCRIPTION
**Pull request by somindatommy**
## Proposed changes in this pull request

Introducing new API definitions to improve NotificationTemplateManager service interface.

Related to issue: https://github.com/wso2/product-is/issues/7006
Related PRs: https://github.com/wso2-extensions/identity-event-handler-notification/pull/114

## Follow-up Actions

After merging this PR https://github.com/wso2-extensions/identity-event-handler-notification/pull/114 has to be merged after updating the governance version.

### CODE REVIEW

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java [hunk 1]
```diff
@@ -34,5 +36,46 @@
...
+    void addNotificationTemplate(NotificationTemplate notificationTemplate, String tenantDomain)
+            throws NotificationTemplateManagerException;
+
+    /**
+     * Add a new notification template to the registry to the corresponding notification channel root directory.
+     *
+     * @param displayName         Notification template display name
+     * @param notificationChannel Notification channel
+     * @param tenantDomain        Tenant domain
+     * @throws NotificationTemplateManagerException Error adding the template to the registry.
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java]: Shall we change this doc comment to one like "Error while adding the template ..."

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java [hunk 2]
```diff
@@ -34,5 +36,46 @@
...
+    void addNotificationTemplateType(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerException;
+
+    /**
+     * Add the default notification templates which matches the given notification channel to the respective tenants
+     * registry.
+     *
+     * @param notificationChannel Notification channel (Eg: SMS, EMAIL)
+     * @param tenantDomain        Tenant domain
+     * @throws NotificationTemplateManagerException Error adding the default notification templates
```
Comment (ShanChathusanda93) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java]: Shall we change this doc comment to one like "Error while adding the template ..."

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java [hunk 3]
```diff
@@ -34,5 +36,46 @@
      * @throws NotificationTemplateManagerException Error while getting the notification template
      */
     NotificationTemplate getNotificationTemplate(String notificationChannel, String templateType, String locale,
-            String tenantDomain) throws NotificationTemplateManagerException;
+                                                 String tenantDomain) throws NotificationTemplateManagerException;
+
+    /**
+     * Add the notification template to the registry.
```
Comment (vihanga-liyanage) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java]: Please format the doc comment. (Try using IntelliJ Idea)

Comment (somindatommy) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java]: Already added and formatted. Therefore, resolving the comment.

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java (hunk 2 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java]: Fixed: 948a27103b94bac935df64e1a165f77d7f4b5b91

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java (hunk 1 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java]: Fixed: 948a27103b94bac935df64e1a165f77d7f4b5b91

**Comment by vihanga-liyanage** [comment]
@somindatommy Please make sure the Travis failure is not because of the PR, before merging.

**Pull request merged**